boards/arm/mimxrt1024_evk/Kconfig.defconfig

```diff
@@ -12,9 +12,6 @@ choice CODE_LOCATION
 	default CODE_FLEXSPI
 endchoice
 
-config FLASH_MCUX_FLEXSPI_NOR
```

Comment:
Since you have changed this to be a choice, this should change to 
```
choice FLEXSPI_FLASH_DRIVER
       default FLASH_MCUX_FLEXSPI_NOR if FLASH
endchoice
```

Reply:
Please read this comment on why this won't work when using an optional choice group.
https://github.com/zephyrproject-rtos/zephyr/pull/38676#discussion_r715897896